src/gpb_compile.erl

```diff
@@ -6156,6 +6156,11 @@ format_msg_record(Msg, Fields, Opts, Defs) ->
      ?f("-endif.~n")].
 
 format_hfields(Indent, Fields, Opts, Defs) ->
+    IsProto3 =
+        case proplists:get_value(syntax, Defs, no) of
+            "proto3" -> true;
+            _ -> false
+        end,
```

Comment:
Use this instead:
```erlang
    IsProto3 = gpb:is_msg_proto3(MsgName, Defs),
```
In a translation unit, it is possible to have a mix of proto2 and proto3 messages. All the parsed items for the .proto and all `import`ed files are concatenated, so there can actually be several `{syntax,...}` items in `Defs`. The first found `syntax` item may be the wrong one.

This applies to one more place below